format_spec/history.md

```diff
@@ -0,0 +1,168 @@
+
+## Version 11
+
+Introduced in TileDB 2.7
+
+* Fragment metadata contain [metadata](./fragment.md#tile-mins-maxes) (min/max value, sum, null count) for each tile.
+
+## Version 10
+
+Introduced in TileDB 2.4
+
+* Arrays support schema evolution.
+    * Array schemas are stored in a `__schema` subdirectory, and have a [timestamped name](./timestamped_name.md).
+    * The _Array schema name_ field was added to the [fragment metadata footer](./fragment.md#footer).
+* The _Footer length_ field of the [fragment metadata footer](./fragment.md#footer) is always written.
+
+## Version 9
+
+Introduced in TileDB 2.3
+
+* [Data files](./fragment.md#data-file) are named by the index of their attribute or dimension.
+* The _URI_ fields of [Consolidated fragment metadata files](./consolidated_fragment_metadata_file.md) contain relative paths to the location of fragments in the array.
+
+## Version 8
+
+Introduced in TileDB 2.2.3
+
+* [Data files](./fragment.md#data-file) are named by the name of their attribute or dimension, after percent encoding certain characters. These characters are `!#$%&'()*+,/:;=?@[]`, as specified in [RFC 3986](https://tools.ietf.org/html/rfc3986), as well as `"<>\|`, which are not allowed in Windows file names.
+
+## Version 7
+
+Introduced in TileDB 2.2
+
+* Attributes can be nullable.
+    * The _Nullable_ and _Fill value validity_ fields were added to [attributes](./array_schema.md#attribute).
+    * The _Validity filters_ field was added to [array schemas](./array_schema.md#array-schema-file).
+    * Fragment metadata contain validity [tile offsets](./fragment.md#tile-offsets).
+
+## Version 6
+
+Introduced in TileDB 2.1
+
+* The _Fill value_ field was added to [attributes](./array_schema.md#attribute).
+
+## Version 5
+
+Introduced in TileDB 2.0
+
+* Dimensions are stored in separate [data files](./fragment.md#data-file).
+* Sparse arrays can have string dimensions and dimensions with different datatypes.
+    * The _Dimension datatype_, _Cell val num_ and _Filters_ fields were added to [dimensions](./array_schema.md#dimension).
+    * The _Domain size_ field was added to [dimensions](./array_schema.md#dimension). The domain of a dimension can have a variable size.
+    * The _Domain datatype_ field was removed from [domains](./array_schema.md#domain).
+    * The [MBR](./fragment.md#mbr) structure has been updated to support variable-sized dimensions.
+    * The _Dimension number_ and _R-Tree datatype_ fields have been removed from [R-Trees](./fragment.md#r-tree).
+* The _Allows dups_ field was added to [array schemas](./array_schema.md#array-schema-file).
+* Committed fragments are indicated by the presence of an `.ok` file in the array's directory, with the same [timestamped name](./timestamped_name.md) as the fragment.
+
+## Version 4
+
+Introduced in TileDB 1.7
+
+> [!NOTE]
+> This version does not bring any changes to supported features in the storage format.
```

Comment:
This version removed KV storage (#1415) but since it's not supported by current TileDB versions, I am inclined to leave KV undocumented.

Reply:
Done.